### ISSUE TITLE
Update DPI/DPR when switching monitors

### ISSUE THREAD

**Issue by wfraser** [jwilm/alacritty#71]
The window size stays the same (post scaling), but the rendered contents stay at the same DPI. When going from low-DPI monitor to a higher-DPI one, this results in the contents not filling the window, and rendering artifacts in the remaining space. Upon resize it adjusts itself to fill correctly. Doing the reverse (high-DPI to lower) results in the contents getting cut off.

Also, because the DPI stays exactly the same, dragging the window from a non-retina monitor to a retina one, the text becomes unreadably tiny instead of scaling up like other windows do.

**Comment by jwilm** [comment]
Thanks for filing the issue! This one will take some time to fix.

**Comment by wfraser** [comment]
Ah, I see, it grabs the `device_pixel_ratio` of whatever monitor it starts up on and just uses it forever. If I start it on the retina one, I get huge text after moving it.

**Comment by trashhalo** [comment]
@jwilm so it looks like you are using [glutin](https://github.com/tomaka/glutin), which is really using [winit](https://github.com/tomaka/winit) for window management. On mac the event you care about is [window​Did​Change​Screen](https://developer.apple.com/reference/appkit/nswindowdelegate/1419267-windowdidchangescreen) which [winit is picking up and and translating it into a window resize event](https://github.com/tomaka/winit/blob/master/src/platform/macos/window.rs#L91) So I think you need to rerequest dpi after getting a resize and you should be good.

**Comment by jwilm** [comment]
@trashhalo That sounds right to me! Thanks for researching this bug. 😄

**Comment by jwilm** [comment]
I should add--a prerequisite would be adding support for font resizing.

**Comment by NickeZ** [comment]
Is this easy to implement once #625 is merged?

**Comment by thomasjachmann** [comment]
For reference: There's the new issue #964 that would fix this.

**Comment by sodiumjoe** [comment]
can you try https://github.com/jwilm/alacritty/pull/1346

**Comment by durin42** [comment]
Works great! Many thanks!
